README.md

```diff
@@ -1,16 +1,61 @@
 # Overview
-This package can be added to your workspaces to provide job runners
-that can fulfill resources requested by plans.
+This package can be used to create a wake `runner` that will set environment
+variables for jobs from JSON values based on the resources they request. A
+default runner is also included with this package.
 
 # Resources
-Resources describe a tool that is requested by a job. They are strings
-that have the format `{vendor}/{tool}/{version}`. If there is no vendor
-then the format is `{tool}/{version}`.
+Resources describe a tool that is requested by a job. They are strings.
+This package expects resources to have the format `{vendor}/{tool}/{version}`.
+If there is no vendor then the format is `{tool}/{version}`.
 
-# Runners
-This package provides a runner that sets the `PATH` of wake
-jobs to the value of the `WAKE_PATH` environment variable if it is set
-by the user.
+# Usage
+To use this you need to create a JSON file that maps resources to a JSON object
+representing the environment variables you want to set when that resource is
+requested.
+```json
+{
+  "google": {
+    "protobuf": {
+      "3.5.1": {
+        "PATH": "/protobuf/bin"
+      }
+    }
+  },
+  "verilator": {
+    "4.008": {
+      "PATH": "/verilator/bin",
+      "VERILATOR_ROOT": "/verilator"
+    }
+  }
+}
+```
 
-This package also provides a runner that will provide the riscv-tools
-resource if the user has set the `RISCV` environment variable.
+Then you need to `publish` a list of `Path`s to your JSON files to `environmentJSONs`.
+```wake
+publish environmentJSONs = source "path/to/environment.json", Nil
+```
+
+Now when a wake job requests the resource `google/protobuf/>=3.5.0`,
```

Comment:
Are we now assuming that version strings will always form a [totally ordered set](https://en.wikipedia.org/wiki/Total_order)? (I.e. all valid version strings can be compared in a "greater than/less than" sense?) We should specify exactly what the parsing rules are for these version numbers, since this can grow in complexity and different languages/systems define different rules beyond the simple case of "period-delimited numbers".

For example, here's Python's version string spec: https://www.python.org/dev/peps/pep-0440/
Here's Maven's: https://maven.apache.org/pom.html#Version_Order_Specification
And Debian's: https://www.debian.org/doc/debian-policy/ch-controlfields.html#version

If we want to keep things simple for now, then I suggest that we explicitly constrain the set of valid version strings to something simple.

Also, we should explicitly define the "operators" that are allowed in the version specifier. How is equality specified? How is a bounded range specified? How are disjoint ranges specified?